File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = []
        for c in s:
            if c == '(' or c == '{' or c == '[':
                st.append(c)
            else:
                if not st or (st[-1] == '(' and c != ')') or (st[-1] == '{' and c != '}') or (st[-1] == '[' and c != ']'):
                    return False
                st.pop()
        return not st

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;
        for(int i=0;i<s.size();i++){
            if(s[i]=='(' || s[i]=='{' || s[i]=='['){
               
                st.push(s[i]);
            }
            else if(s[i]==')'){
                if(empty(st) || st.top()!='(') return false;
                st.pop();
            }
            else if(s[i]=='}'){
                if(empty(st) || st.top()!='{') return false;
                st.pop();
            }
            else if(s[i]==']'){
                if(empty(st) || st.top()!='[') return false;
                st.pop();
            }
        }
        if(empty(st)) return true;
        return false;
    }
};